src/actions/clone.js

```diff
@@ -103,26 +103,27 @@ async function cloneFile(lock, ctx, params) {
   // add mark to the original file
   pipeline.hset(uploadKey, FILES_HAS_CLONES_FIELD, 1);
 
-  const provider = ctx.provider('remove', params, undefined, uploadData);
+  const prefix = md5(username);
+  const copyPromises = JSON.parse(mergedData.files).map(async (file) => {
+    const provider = ctx.provider('remove', uploadData, file);
 
-  if (provider.canCopy()) {
-    const prefix = md5(username);
+    if (!provider.canCopy()) {
```

Comment:
Cloudflare Stream API has no method for copy file, but it might make sense to upload videos using "Upload via link" method:
https://developers.cloudflare.com/api/operations/stream-videos-upload-videos-from-a-url
https://developers.cloudflare.com/stream/uploading-videos/upload-via-link/